File: core/src/main/java/in/koreatech/koin/core/toast/ToastUtil.java
package in.koreatech.koin.core.toast;

import android.content.Context;
import android.widget.Toast;

/**
 * @author hyerim
 * @author nayunjae
 * @since 2018.4.9
 */
public class ToastUtil {
    private static Context applicationContext;

    private ToastUtil() {
    }

    private static class Holder {
        private static final ToastUtil toastUtil = new ToastUtil();
    }

    public static ToastUtil getInstance() {
        return Holder.toastUtil;
    }

    public void init(Context context) {
        applicationContext = context.getApplicationContext();
    }

    public void makeShort(String message) {
        Toast.makeText(applicationContext, message, Toast.LENGTH_SHORT).show();
    }

    public void makeLong(String message) {
        Toast.makeText(applicationContext, message, Toast.LENGTH_LONG).show();
    }

    public void makeShort(int message) {
        Toast.makeText(applicationContext, message, Toast.LENGTH_SHORT).show();
    }

    public void makeLong(int message) {
        Toast.makeText(applicationContext, message, Toast.LENGTH_LONG).show();
    }
}

File: core/src/main/java/in/koreatech/koin/core/toast/ToastUtil.kt
package `in`.koreatech.koin.core.toast

import android.content.Context
import android.widget.Toast

object ToastUtil {
    lateinit var applicationContext: Context

    @Deprecated("Access ToastUtil functions directly")
    @JvmStatic
    val instance = this

    fun init(context: Context) {
        applicationContext = context.applicationContext
    }

    fun makeShort(message: String?) {
        Toast.makeText(applicationContext, message, Toast.LENGTH_SHORT).show()
    }

    fun makeLong(message: String?) {
        Toast.makeText(applicationContext, message, Toast.LENGTH_LONG).show()
    }

    fun makeShort(message: Int) {
        Toast.makeText(applicationContext, message, Toast.LENGTH_SHORT).show()
    }

    fun makeLong(message: Int) {
        Toast.makeText(applicationContext, message, Toast.LENGTH_LONG).show()
    }
}